shaderc-rs/src/lib.rs

```diff
@@ -708,14 +737,15 @@ impl<'a> CompileOptions<'a> {
     /// * Target environment: Vulkan
     /// * Source language: GLSL
     ///
-    /// A return of `None` indicates that there was an error initializing
-    /// the underlying options object.
-    pub fn new() -> Option<CompileOptions<'a>> {
+    /// Returns `Err(Error::InitializationError)` if creation failed.
+    pub fn new() -> Result<CompileOptions<'a>> {
         let p = unsafe { scs::shaderc_compile_options_initialize() };
         if p.is_null() {
-            None
+            Err(Error::InitializationError(
+                "Failed to create CompileOptions.".to_string(),
```

Comment:
Similarly here: use lower case and drop ending `.`

Reply:
Thank you!